DEVELOPMENT.md

```diff
@@ -0,0 +1,242 @@
+  before proceeding.
+
+## Micro-RDK Development Tips
+
+### Tree-level `cargo` configuration
+
+The micro-rdk monorepo is only part of the development story: when
+developing for or with Micro-RDK, there will be other directories in
+play (e.g. projects generated from the template, etc.), and
+coordinating the state of these many directories and repositories can
+prove troublesome. A carefully placed `.cargo/config.toml` file can
+inject settings for the entire tree of trees, including the micro-rdk
+monorepo, projects, and modules. Simply create a `.cargo` directory as
+a peer of your micro-rdk monorepo and any project directories, and
+then create a `config.toml` file within it. Settings applied in that
+file will now apply globally to all crates inside the micro-rdk as
+well as to peer project directories. Here is an example layout:
+
+```
+.
+├── .cargo
+│   └── config.toml
+├── micro-rdk
+│   ├── ...
+│   ├── examples
+│   ├── micro-rdk
+│   ├── micro-rdk-ffi
+│   ├── micro-rdk-installer
+│   ├── micro-rdk-macros
+│   ├── micro-rdk-server
+│   └── templates
+└── projects
+    ├── module-under-development
+    ├── another-module
+    └── module-consuming-project
+```
+
+Settings made to the `.cargo/config.toml` file will now apply
+uniformly to both the `micro-rdk` tree, as well as to the projects
+under the `projects` directory. Several of the remaining development
+tips in this document make use of this functionality.
+
+### Using `sccache` to improve build times
+
+Use [sccache](https://github.com/mozilla/sccache) to cache the result
+of rust compiles for an overall faster experience as you move back and
+forth between branches. Install it with `cargo install sccache` and
+then you can opt into it by setting the `RUSTC_WRAPPER` environment
+variable to point to `sccache`, e.g. `RUSTC_WRAPPER=/path/to/sccache
+cargo build ...`. But it can be easy to forget to configure
+that. Instead, you can add the following to a `.cargo/config.toml`
+(either in `$HOME/.cargo/config.toml` to apply it everywhere, or in a
+tree-level `.cargo/config.toml` per above):
+
+
+```
+[build]
+rustc-wrapper = "/path/to/sccache"
+[profile.dev.package."*"]
+incremental = false
+```
+
+Note that this disables incremental compilation for dev builds because
+`sccache` is incompatible with incremental compilation.
+
+### Fixing `+esp` Builds on MacOS
+
+If you are building `Micro-RDK` for ESP32 on a macOS machine and you
+receive an error like the following:
+
+```
+xtensa-esp32-elf-gcc: error: unrecognized command line option '--target=xtensa-esp32-espidf'
+```
+
+Then [work around](https://github.com/esp-rs/esp-idf-template/issues/174) this
+issue by setting `CRATE_CC_NO_DEFAULTS=1` in your environment
+(e.g. `make build-esp32-bin CRATE_CC_NO_DEFAULTS=1` or
+`CRATE_CC_NO_DEFAULTS=1 cargo +esp build ...`. Or, you can use the
+tree-level Cargo configuration to just take care of this for you:
+
+```
```

Comment:
specify that this goes in `.cargo/config.toml`?

Reply:
Done